src/Compilers/CSharp/Test/Emit3/Symbols/UserDefinedCompoundAssignmentOperatorsTests.cs

```diff
@@ -13931,6 +13931,286 @@ .locals init (int V_0,
+            CreateCompilation([source, CompilerFeatureRequiredAttribute]).VerifyDiagnostics(
+                // (6,16): error CS8350: This combination of arguments to 'C.operator +=(C)' is disallowed because it may expose variables referenced by parameter 'right' outside of their declaration scope
+                //         return c += c1;
+                Diagnostic(ErrorCode.ERR_CallArgMixing, "c += c1").WithArguments("C.operator +=(C)", "right").WithLocation(6, 16),
+                // (6,21): error CS8352: Cannot use variable 'scoped C c1' in this context because it may expose referenced variables outside of their declaration scope
+                //         return c += c1;
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "c1").WithArguments("scoped C c1").WithLocation(6, 21)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignment_00749_Consumption_RefSafety()
+        {
+            var source = """
+                public ref struct C
+                {
+                    public void operator +=(C right) {}
+                    public C M1(scoped C c, scoped C c1)
+                    {
+                        return c += c1;
+                    }
+                }
+                """;
+            CreateCompilation([source, CompilerFeatureRequiredAttribute]).VerifyDiagnostics(
+                // (6,16): error CS8352: Cannot use variable 'scoped C c' in this context because it may expose referenced variables outside of their declaration scope
+                //         return c += c1;
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "c").WithArguments("scoped C c").WithLocation(6, 16)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignment_00750_Consumption_RefSafety()
+        {
+            var source = """
+                public ref struct C
+                {
+                    public void operator +=(C right) {}
+                    static C X(C c) => throw null;
+                    public C M1(scoped C c, scoped C c1)
+                    {
+                        return X(c += c1);
+                    }
+                }
+                """;
+            CreateCompilation([source, CompilerFeatureRequiredAttribute]).VerifyDiagnostics(
+                // (7,16): error CS8347: Cannot use a result of 'C.X(C)' in this context because it may expose variables referenced by parameter 'c' outside of their declaration scope
+                //         return X(c += c1);
+                Diagnostic(ErrorCode.ERR_EscapeCall, "X(c += c1)").WithArguments("C.X(C)", "c").WithLocation(7, 16),
+                // (7,18): error CS8352: Cannot use variable 'scoped C c' in this context because it may expose referenced variables outside of their declaration scope
+                //         return X(c += c1);
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "c").WithArguments("scoped C c").WithLocation(7, 18)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are we testing by-ref parameters? Consider testing the following, with/without `[UnscopedRef]`:
```csharp
using System.Diagnostics.CodeAnalysis;
ref struct C
{
    private ref readonly int _i;
    public void operator +=([UnscopedRef] in int right) { _i = ref right; }
    public C M1(C c)
    {
        return c += 1;
    }
}
```

Reply:
Added CompoundAssignment_00752_Consumption_RefSafety and CompoundAssignment_00753_Consumption_RefSafety